web/src/people.ts

```diff
@@ -14,15 +14,66 @@ import * as settings_config from "./settings_config";
 import * as settings_data from "./settings_data";
 import * as util from "./util";
 
-let people_dict;
-let people_by_name_dict;
-let people_by_user_id_dict;
-let active_user_dict;
-let non_active_user_dict;
-let cross_realm_dict;
-let pm_recipient_count_dict;
-let duplicate_full_name_data;
-let my_user_id;
+export type ProfileData = {
+    value: string;
+    rendered_value: string;
+};
+
+export type User = {
+    user_id: number;
+    delivery_email?: string;
+    email: string;
+    full_name: string;
+    date_joined: string;
+    is_active: boolean;
+    is_owner: boolean;
+    is_admin: boolean;
+    is_guest: boolean;
+    is_moderator?: boolean;
+    is_billing_admin: boolean;
+    is_bot: boolean;
+    bot_type?: number;
+    bot_owner_id?: number;
```

Comment:
Both `bot_type` and `bot_owner_id` are also nullable.

Reply:
Note that the server has an undocumented behavior that `bot_type` and `bot_owner_id` do not appear when the user is not `null`. We could probably work to make that more accurately documented/behave consistently, but that will be a task of its own.
https://github.com/zulip/zulip/blob/main/zerver/lib/users.py#L501-L507